nomad/structs/volumes.go

```diff
@@ -0,0 +1,188 @@
+package structs
+
+import (
+	"github.com/mitchellh/copystructure"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	VolumeTypeHost = "host"
+)
+
+// ClientHostVolumeConfig is used to configure access to host paths on a Nomad Client
+type ClientHostVolumeConfig struct {
+	Name     string `hcl:",key"`
+	Source   string `hcl:"source"`
+	ReadOnly bool   `hcl:"read_only"`
+	Hidden   bool   `hcl:"hidden"`
+}
+
+func (p *ClientHostVolumeConfig) Copy() *ClientHostVolumeConfig {
+	if p == nil {
+		return nil
+	}
+
+	c := new(ClientHostVolumeConfig)
+	*c = *p
+	return c
```

Comment:
I think you can get away with:

```
	c := *p
	return &c
```

Reply:
Probably, but we use the same style for `Copy` implementations in most of nomad, so differences from it call out a lot of attention imo